awaitility/src/main/java/com/jayway/awaitility/core/ConditionAwaiter.java

```diff
@@ -164,6 +165,20 @@ public void uncaughtException(Thread thread, Throwable throwable) {
         }
     }
 
+    private ExecutorService initExecutorService() {
+        return Executors.newSingleThreadExecutor(new ThreadFactory() {
+            public Thread newThread(Runnable r) {
+                Thread thread;
+                if (conditionSettings.hasAlias()) {
+                    thread = new Thread(Thread.currentThread().getThreadGroup(), r, "awaitility[" + conditionSettings.getAlias() + ']');
+                } else {
+                    thread = new Thread(Thread.currentThread().getThreadGroup(), r, "awaitility-thread");
```

Comment:
I'm not sure whether the default naming "pool-3-thread-2" is any better than generic "awaitility-thread". You can't distinguish them one from another. Its no better than having them named the same.